demosplan/DemosPlanCoreBundle/ResourceTypes/StatementVoteResourceType.php

```diff
@@ -0,0 +1,179 @@
+                [StatementInterface::PUBLICATION_PENDING, StatementInterface::PUBLICATION_APPROVED, StatementInterface::PUBLICATION_REJECTED],
+                Paths::statementVote()->statement->publicVerified)
+        );
+
+        $statementVoteConfig = $this->getConfig(StatementVoteResourceConfigBuilder::class);
+
+        $statementVoteConfig->id->setReadableByPath();
+
+        $statementVoteConfig->name
+            ->readable(true, fn (StatementVote $statementVote): string => $statementVote->getName())
+            ->updatable([$voteConditions], function (StatementVote $statementVote, ?string $name): array {
+                $statementVote->setLastName($name);
+
+                return [];
+            })
+            ->initializable(true, static function (StatementVote $statementVote, ?string $name): array {
+                $statementVote->setLastName($name);
+
+                return [];
+            });
+
+        $statementVoteConfig->email
+            ->setReadableByPath()
+            ->addPathUpdateBehavior([$voteConditions])
+            ->addPathCreationBehavior(OptionalField::YES, [$voteConditions])
+            ->setAliasedPath(Paths::statementVote()->userMail);
+
+        $statementVoteConfig->city
+            ->setReadableByPath()
+            ->addPathUpdateBehavior([$voteConditions])
+            ->addPathCreationBehavior(OptionalField::YES, [$voteConditions])
+            ->setAliasedPath(Paths::statementVote()->userCity);
+
+        $statementVoteConfig->postcode
+            ->setReadableByPath()
+            ->addPathUpdateBehavior([$voteConditions])
+            ->addPathCreationBehavior(OptionalField::YES, [$voteConditions])
+            ->setAliasedPath(Paths::statementVote()->userPostcode);
+
+        $statementVoteConfig->createdByCitizen
+            ->setReadableByPath()
+            ->addPathUpdateBehavior([$voteConditions])
+            ->addPathCreationBehavior(OptionalField::NO, [$voteConditions]);
+
+        $statementVoteConfig->organisationName
+            ->setReadableByPath()
+            ->addPathUpdateBehavior([$voteConditions])
+            ->addPathCreationBehavior(OptionalField::YES, [$voteConditions]);
+
+        $statementVoteConfig->departmentName
+            ->setReadableByPath()
+            ->addPathUpdateBehavior([$voteConditions])
+            ->addPathCreationBehavior(OptionalField::YES, [$voteConditions]);
+
+        $statementVoteConfig->user
+            ->setRelationshipType($this->resourceTypeStore->getUserResourceType())
+            ->setReadableByPath();
+
+        $statementVoteConfig->statement
+            ->setRelationshipType($this->resourceTypeStore->getStatementResourceType())
+           // ->addConstructorBehavior(ToOneRelationshipConstructorBehavior::createFactory(null, [], null, OptionalField::NO))
+            ->initializable(false, static function (StatementVote $statementVote, Statement $statement): array {
```

Comment:
also deprecated?

Reply:
@breierDemos I've been trying to adjust the initializable behavior using the new EDT version. After several attempts that resulted in errors, I decided to use this approach to get it working. I will create another ticket related to upgrading to the newest EDT version and we can proceed to merge this PR

Reply:
:+1: thanks